cdxev/amend/operations.py

```diff
@@ -228,3 +229,35 @@ def handle_component(self, component: dict) -> None:
         process_license(
             component, self.list_of_license_names, self.path_to_license_folder
         )
+
+
+class InferCopyright(Operation):
+    """
+    If neither a license nor a copyright is present in a component,
+    this function will create a 'copyright' field in the schema
+    'supplier.name year, all rights reserved'
+    """
+
+    def infer_copyright(self, component: dict) -> None:
+        if "copyright" in component.keys() or "licenses" in component.keys():
+            return
+
+        if "supplier" not in component.keys():
+            return
+
+        if "name" not in component.get("supplier", {}).keys():
+            return
+
+        year = datetime.date.today().year
+        supplier_name = component.get("supplier", {}).get("name", "")
+        copyright = f"Copyright {year} {supplier_name}, all rights reserved"
```

Comment:
I would remove the "all rights reserved", there are some [discussions](https://opensource.stackexchange.com/questions/2121/mit-license-and-all-rights-reserved/4403#4403) related to this topic. It seems to have no significance as it is the [default](https://www.iusmentis.com/copyright/allrightsreserved/). I would use the text "Copyright (c) {year} {supplier_name}".

Reply:
Removed